src/Packetery/Module/Order/LabelPrint.php

```diff
@@ -180,20 +181,34 @@ public function outputLabelsPdf(): void {
 		$isCarrierLabels        = ( $this->httpRequest->getQuery( self::LABEL_TYPE_PARAM ) === self::ACTION_CARRIER_LABELS );
 		$orderIdParam           = $this->httpRequest->getQuery( 'id' ) !== null ? (int) $this->httpRequest->getQuery( 'id' ) : null;
 		$packetIdParam          = $this->httpRequest->getQuery( 'packet_id' );
-		if ( $orderIdParam !== null && $packetIdParam !== null ) {
+		if ( $orderIdParam !== null && $packetIdParam !== null && is_string( $packetIdParam ) ) {
 			$fallbackToPacketaLabel = true;
-			$packetIds              = [ $orderIdParam => $packetIdParam ];
+			$order                  = $this->orderRepository->getByIdWithValidCarrier( $orderIdParam );
+			if ( $order === null ) {
+				$this->messageManager->flash_message(
```

Comment:
Pro toto chování (nová hláška + redirect) nevidím podklady v zadání ani analýze. Je to schváleno?